runtime/src/crowdfund.rs

```diff
@@ -0,0 +1,547 @@
+//! (efficient) redistribution back to the contributors.
+//!
+//! Contributions must be of at least `MinContribution` (to account for the resources taken in
+//! tracking contributions), and may never tally greater than the fund's `cap`, set and fixed at the
+//! time of creation. The `create` call may be used to create a new fund. In order to do this, then
+//! a deposit must be paid of the amount `SubmissionDeposit`. Substantial resources are taken on
+//! the main trie in tracking a fund and this accounts for that.
+//!
+//! Funds may be set up during an auction period; their closing time is fixed at creation (as a
+//! block number) and if the fund is not successful by the closing time, then it will become *retired*.
+//! Contributors may get a refund of their contributions from retired funds. After a period (`RetirementPeriod`)
+//! the fund may be dissolved entirely. At this point any non-refunded contributions are considered
+//! `orphaned` and are disposed of through the `OrphanedFunds` handler (which may e.g. place them
+//! into the treasury).
+//!
+//! Funds may accept contributions at any point before their success or retirement. When a parachain
+//! slot auction enters its ending period, then parachains will each place a bid; the bid will be
+//! raised once per block if the parachain had additional funds contributed since the last bid.
+//!
+//! Funds may set their deploy data (the code hash and head data of their parachain) at any point.
+//! It may only be done once and once set cannot be changed. Good procedure would be to set them
+//! ahead of receiving any contributions in order that contributors may verify that their parachain
+//! contains all expected functionality. However, this is not enforced and deploy data may happen
+//! at any point, even after a slot has been successfully won or, indeed, never.
+//!
+//! Funds that are successful winners of a slot may have their slot claimed through the `onboard`
+//! call. This may only be done once and must be after the deploy data has been fixed. Successful
+//! funds remain tracked (in the `Funds` storage item and the associated child trie) as long as
+//! the parachain remains active. Once it does not, it is up to the parachain to ensure that the
+//! funds are returned to this module's fund sub-account in order that they be redistributed back to
+//! contributors. *Retirement* may be initiated by any account (using the `begin_retirement` call)
+//! once the parachain is removed from the its slot.
+//!
+//! @WARNING: For funds to be returned, it is imperative that this module's account is provided as
+//! the offboarding account for the slot. In the case that a parachain supplemented these funds in
+//! order to win a later auction, then it is the parachain's duty to ensure that the right amount of
+//! funds ultimately end up in module's fund sub-account. If the funds do not arrive, then
```

Comment:
```suggestion
//! funds ultimately end up in module's fund sub-account.
```

I assume the hanging last sentence might have read: "If the funds do not arrive, then the parachain will not be able to bid on future auctions with the funds." This sentence would be redundant given the previous sentence.